src/main/java/com/cloudbees/jenkins/support/impl/ThreadDumps.java

```diff
@@ -273,12 +273,16 @@ public static void printThreadInfo(PrintWriter writer, ThreadInfo t, ThreadMXBea
             x.printStackTrace(writer);
             cpuPercentage = 0;
         }
-        writer.printf("\"%s\" id=%d (0x%x) state=%s cpu=%d%%",
+
+        long time = t.getBlockedTime() + t.getWaitedTime();
+
+        writer.printf("\"%s\" id=%d (0x%x) state=%s cpu=%d%% time=%dms",
                 t.getThreadName(),
                 t.getThreadId(),
                 t.getThreadId(),
                 t.getThreadState(),
-                cpuPercentage);
+                cpuPercentage,
```

Comment:
cpuPercentage is a useless metric - it is the metric of `(now - t1)/uptime)` where `t1` is the time at which the thread was started (so all threads started with Jenkins will show as 100%, even if they have been completely idle for the whole period.